nucypher/blockchain/eth/actors.py

```diff
@@ -100,25 +98,26 @@ def __init__(self, blockchain: BlockchainInterface, checksum_address: str = None
         except AttributeError:
             self.checksum_address = checksum_address  # type: str
 
-        self.blockchain = blockchain
-        self.token_agent = NucypherTokenAgent(blockchain=self.blockchain)
-        self._saved_receipts = list()  # type: list # track receipts of transmitted transactions
+        self.registry = registry
+        self.token_agent = ContractAgency.get_agent(NucypherTokenAgent, registry=self.registry)
+        self._saved_receipts = list()  # track receipts of transmitted transactions
 
     def __repr__(self):
         class_name = self.__class__.__name__
-        r = "{}(address='{}')"
+        r = "{}(staker_address='{}')"
```

Comment:
Not all `NucypherTokenActors` are stakers.

Reply:
Agreed.

Reply:
:heavy_check_mark: - OK - weve reverted the over-naming.